worker/metrics/collect/manifold.go

```diff
@@ -265,9 +266,15 @@ func (h *hookRunner) do(recorder spool.MetricRecorder) error {
 	}
 
 	r := runner.NewRunner(ctx, h.paths, nil)
-	_, err = r.RunHook(string(hooks.CollectMetrics))
-	if err != nil {
-		return errors.Annotatef(err, "error running 'collect-metrics' hook")
+	handlerType, err := r.RunHook(string(hooks.CollectMetrics))
+	switch {
+	case err == nil && handlerType == runner.InvalidHookHandler, charmrunner.IsMissingHookError(errors.Cause(err)):
```

Comment:
This is interesting, didn't know that you could use the comma syntax with a generic switch! 

I think it would look cleaner like:

```
case charmrunner.IsMissingHookError(errors.Cause(err)) || (err == nil && handlerType == runner.InvalidHookHandler)
```

or

```
case charmrunner.IsMissingHookError(errors.Cause(err)):
    fallthrough
case (err == nil && handlerType == runner.InvalidHookHandler):
```

Reply:
Yes, the comma works great here especially with single word items.  Agreed, it's harder to read in this case.  Making 2 case statements.